### PR TITLE
CO-156: hard-gate shared Linear request headroom

### PR DESCRIPTION
**Pull request by Kbediako**
## Summary
- hard-gate shared Linear request headroom across provider/control-host Linear consumers
- persist bounded request-burn history with run/process attribution
- limit reserve overrides to one truthful helper read and fail paginated follow-on reads closed

## What Changed
- added reserve-aware shared-budget preflight and request-burn history persistence in `linearBudgetState`
- threaded reserve-aware gating through provider helper mutations while preserving a single essential live read where needed
- added regression coverage for reserve-aware helper behavior, dispatch gating, burn-history persistence, and first-page-only reserve overrides
- refreshed the required docs-first packet for CO-156

## Validation
- `node scripts/delegation-guard.mjs`
- `node scripts/spec-guard.mjs --dry-run`
- `npm run build`
- `npm run lint`
- `npm run test`
- `npm run docs:check`
- `npm run docs:freshness`
- `npm run repo:stewardship`
- `DIFF_BUDGET_OVERRIDE_REASON='CO-156 requires a docs-first packet plus cross-cutting shared Linear budget governor and regression coverage in one auditable issue slice.' node scripts/diff-budget.mjs`
- `FORCE_CODEX_REVIEW=1 CODEX_REVIEW_NON_INTERACTIVE=1 ... npm run review -- --manifest ../../.runs/linear-e934e6ea-8fdd-480c-9656-22c020cb19cf/cli/2026-04-12T02-37-24-809Z-79de77b3/manifest.json` (`review_outcome: bounded-success`)
- `npm run pack:smoke`

## Notes
- merged the latest `origin/main` into the branch before handoff
- `docs/TASKS.md` needed a one-task archive after the main merge to restore `docs:check` budget compliance
- untracked `.child-lanes/` artifacts were left local and are not part of this PR

## Summary by CodeRabbit

* **New Features**
  * Hard-gate nonessential Linear consumers when shared request headroom falls below the configured reserve.
  * Persist request-burn telemetry to track quota usage over time with source and operation attribution.

* **Tests**
  * Expanded coverage validating reserve enforcement, fail-fast behaviours, expiry/merge edge-cases and telemetry persistence.

* **Documentation**
  * Added PRD, TECH_SPEC, ACTION_PLAN, task checklists and registry updates capturing design, validation and rollout steps.

Diff budget override: CO-156 requires a docs-first packet plus cross-cutting shared Linear budget governor and regression coverage in one auditable issue slice.

### CODE REVIEW

File: orchestrator/src/cli/control/linearBudgetState.ts [hunk 1]
```diff
@@ -368,6 +412,35 @@ export function resolveLinearBudgetPreflight(input: {
...
   }
 
+  if (input.allow_below_request_reserve !== true) {
+    const requestReserveShortfall = resolveRequestReserveShortfall(
+      [
+        ['requests', budget.requests],
+        ['endpoint_requests', budget.endpoint_requests]
+      ],
+      minimumRequestsRemaining
+    );
```
[the remark this answers is not in the document]
Comment (Kbediako) [reply orchestrator/src/cli/control/linearBudgetState.ts:422]: Adjusted the unknown-reset reserve path in `linearBudgetState.ts` so fresh unknown-reset buckets still fail closed, but stale reserve-floor buckets decay through hydration after the existing fallback grace window instead of wedging indefinitely. Added regression coverage in `LinearBudgetState.test.ts` for the stale unknown-reset reserve-floor case. Pushed in `aabbaf8fa`.

File: .agent/task/linear-e934e6ea-8fdd-480c-9656-22c020cb19cf.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+- [ ] `node scripts/diff-budget.mjs`. Evidence: pending.
+- [ ] `npm run review`. Evidence: pending.
+- [ ] `npm run pack:smoke`. Evidence: pending.
+
+## Handoff
+- [ ] Workpad refreshed with docs-review evidence, implementation scope, and current validation truth. Evidence: pending.
+- [ ] PR attached to the Linear issue before review-state transition. Evidence: pending.
+- [ ] Latest `origin/main` merged into the branch before review-state transition. Evidence: pending.
+- [ ] PR checks green and actionable review feedback handled or explicitly pushed back before review-state transition. Evidence: pending.
+- [ ] Issue moved to the actual team review state (`In Review`) only after coding stops. Evidence: pending.
```
[the remark this answers is not in the document]
Comment (Kbediako) [reply .agent/task/linear-e934e6ea-8fdd-480c-9656-22c020cb19cf.md]: Updated both checklist mirrors in `0352c4ade` so completed implementation/validation rows are now `[x]` with concrete evidence paths, the `.agent` mirror line points at the paired `tasks/tasks-...` file instead of itself, and still-pending gates stay pending (for example the docs-review child stream and final review-state transition).

File: orchestrator/src/cli/control/linearBudgetState.ts [hunk 3]
```diff
@@ -701,7 +776,16 @@ async function recordLinearBudgetObservation(input: {
...
+    const mergedWithoutHistory = mergePersistedLinearBudgetStatus(existing, observation, scope);
+    const merged = appendPersistedLinearBudgetRequestBurnHistory(
+      mergedWithoutHistory,
+      buildPersistedLinearBudgetRequestBurnHistoryEntry({
+        env,
+        existing,
+        observation,
+        merged: mergedWithoutHistory
+      })
+    );
```
[the remark this answers is not in the document]
Comment (Kbediako) [reply orchestrator/src/cli/control/linearBudgetState.ts:791]: Fixed in `0352c4ade`. `recordLinearBudgetObservation(...)` now skips burn-history append when the incoming observation is stale enough that `mergePersistedLinearBudgetStatus(...)` keeps the newer persisted snapshot, and `LinearBudgetState.test.ts` now covers the rejected-stale-observation path so we do not emit inverted `remaining_delta` telemetry.

File: tasks/tasks-linear-e934e6ea-8fdd-480c-9656-22c020cb19cf.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+- [ ] `node scripts/diff-budget.mjs`. Evidence: pending.
+- [ ] `npm run review`. Evidence: pending.
+- [ ] `npm run pack:smoke`. Evidence: pending.
+
+## Handoff
+- [ ] Workpad refreshed with docs-review evidence, implementation scope, and current validation truth. Evidence: pending.
+- [ ] PR attached to the Linear issue before review-state transition. Evidence: pending.
+- [ ] Latest `origin/main` merged into the branch before review-state transition. Evidence: pending.
+- [ ] PR checks green and actionable review feedback handled or explicitly pushed back before review-state transition. Evidence: pending.
+- [ ] Issue moved to the actual team review state (`In Review`) only after coding stops. Evidence: pending.
```
[the remark this answers is not in the document]
Comment (Kbediako) [reply tasks/tasks-linear-e934e6ea-8fdd-480c-9656-22c020cb19cf.md:42]: Reconciled the `tasks/tasks-...` checklist in `0352c4ade` to match execution truth: implementation, validation, review, pack-smoke, PR attachment, and workpad refresh now point at concrete logs/telemetry/workpad evidence, while the still-open handoff gates remain pending until the PR drain and review-state transition actually complete.
